src/foam/mlang/mlang.js

```diff
@@ -1592,6 +1592,35 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
+  package: 'foam.mlang.sink',
+  name: 'Min',
+  extends: 'foam.dao.AbstractSink',
+
+  implements: [
+    'foam.mlang.predicate.Unary',
+    'foam.core.Serializable'
+  ],
+
+  documentation: 'A Sink which remembers the minimum value put().',
+
+  properties: [
+    {
+      name: 'value',
+      value: 0
+    }
+  ],
+
+  methods: [
+    function put(obj) {
+      if ( ! this.hasOwnProperty('value') ) {
```

Comment:
If you OR'ed the two if clauses then you wouldn't need to duplicate the body twice.